cores/arduino/stm32/twi.c

```diff
@@ -512,9 +527,14 @@ void HAL_I2C_ListenCpltCallback(I2C_HandleTypeDef *hi2c)
   */
 void HAL_I2C_ErrorCallback(I2C_HandleTypeDef *hi2c)
 {
-  HAL_I2C_EnableListen_IT(hi2c);
+  i2c_t *obj = get_i2c_obj(hi2c);
+
+  if(obj->isMaster == 0) {
+    HAL_I2C_EnableListen_IT(hi2c);
```

Comment:
can you comment or explain this ErrorCallback management ? 
If there is an error, shouldn't we report the error to application ?

Reply:
In master mode, the callback is useless because the error is reported to the Arduino API with `i2c_master_write()` and `i2c_master_read()`.
In slave mode, there is no mechanism in Arduino API to report an error so the error callback forces the slave to listen again.

Reply:
Ok Good. So please explain this either with comments in the code, or in the commit message, this is useful information that we cannot guess when reading the code

Reply:
Done!

Reply:
where is it done ?

Reply:
Above in the header of the function.